src/toil_scripts/lib/urls.py

```diff
@@ -1,32 +1,31 @@
-import base64
 import glob
-import hashlib
 import os
+import shutil
 import subprocess
 from urlparse import urlparse
-import shutil
+
+from toil_scripts.lib import require
 from toil_scripts.lib.programs import docker_call
 
 
-def download_url(url, work_dir='.', name=None, s3_key_path=None, cghub_key_path=None):
+def download_url(url, work_dir='.', name=None, num_cores=1, s3_key_path=None, cghub_key_path=None):
     """
     Downloads URL, can pass in file://, http://, s3://, or ftp://, gnos://cghub/analysisID, or gnos:///analysisID
 
     :param str url: URL to download from
     :param str work_dir: Directory to download file to
     :param str name: Name of output file, if None, basename of URL is used
+    :param int num_cores: Number of cores to use if downloading with s3am
     :param str s3_key_path: Path to 32-byte encryption key if url points to S3 file that uses SSE-C
     :param str cghub_key_path: Path to cghub key used to download from CGHub.
     :return: Path to the downloaded file
     :rtype: str
     """
     file_path = os.path.join(work_dir, name) if name else os.path.join(work_dir, os.path.basename(url))
-    if s3_key_path:
-        _download_encrypted_file(url, file_path, s3_key_path)
-    elif cghub_key_path:
-        _download_from_genetorrent(url, file_path, cghub_key_path)
+    if cghub_key_path:
+        _download_with_genetorrent(url, file_path, cghub_key_path)
     elif urlparse(url).scheme == 's3':
-        _download_s3_url(file_path, url)
+        _s3am_with_retry(num_cores, file_path, s3_url=url, s3_key_path=s3_key_path, mode='download')
```

Comment:
Why? Then i'd have to supply defaults for file_path and s3_url which doesn't make sense.